sdk/identity/azure_identity/src/credentials/azure_cli_credentials.rs

```diff
@@ -131,19 +37,10 @@ impl CliTokenResponse {
                 .with_context(ErrorKind::DataConversion, || {
                     format!("unable to parse expires_on '{timestamp}'")
                 })?),
-            None => {
-                #[cfg(feature = "old_azure_cli")]
-                {
-                    Ok(self.local_expires_on)
-                }
-                #[cfg(not(feature = "old_azure_cli"))]
-                {
-                    Err(Error::message(
-                        ErrorKind::DataConversion,
-                        "expires_on field not found. Please use Azure CLI 2.54.0 or newer.",
-                    ))
```

Comment:
We should leave something to this effect. I worry about sudden breakage for those that installed `az` long ago and haven't updated since.

Reply:
I removed the feature macros, not this error; it's now returned from the None arm in any case